dbms/src/Functions/FunctionsTiDBConversion.h

```diff
@@ -1610,6 +1611,47 @@ struct TiDBConvertToDuration
                 block.getByPosition(result).column = std::move(to_col);
             }
         }
+        else if constexpr (std::is_same_v<FromDataType, DataTypeMyDate> || std::is_same_v<FromDataType, DataTypeMyDateTime>)
+        {
```

Comment:
can you split the execute function by FromDataType?

Reply:
done